### ISSUE TITLE
DateUtils.isToday() returning incorrect result.

### ISSUE THREAD

**Issue by evant** [robolectric/robolectric#3912]
### Description
Getting false when I run `DateUtils.isToday(System.currentTimeMillis())` in a test.

### Robolectric & Android Version
robolectric: 4.0-alpha-1
AGP: 3.20-alpha14

**Comment by cezium** [comment]
This is still a problem

com.android.tools.build:gradle:3.2.1
org.robolectric:robolectric:4.0.2

`package com.test.brokendate

import android.os.Build
import android.text.format.DateUtils
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config
import java.util.Date
import java.util.Locale


@RunWith(RobolectricTestRunner::class)
@Config(sdk = [Build.VERSION_CODES.O_MR1])
class DateUtilTest {


    @Test
    fun testNow() {
        val locale = Locale("sv", "se")
        Locale.setDefault(locale)
        Assert.assertEquals(true, DateUtils.isToday(Date().getTime()))
    }
}`

**Comment by PascalLzx** [comment]
`        long today = java.util.Calendar.getInstance().getTimeInMillis();
        ShadowSystemClock.setCurrentTimeMillis(today);
        Assert.assertEquals(true, DateUtils.isToday(today));`

**Comment by St4B** [comment]
@PascalLzx 's workaround does not seem to work anymore :S

**Comment by St4B** [comment]
Hmmm... It seems to be solved but not yet released 
https://github.com/robolectric/robolectric/pull/5075

**Comment by spacifici** [comment]
@St4B 

> Hmmm... It seems to be solved but not yet released
> #5075

Nope, it doesn't the problem is different. Can you please try if running with a regular JVM (not the one distributed with Android Studio) resolve the exception?

**Comment by caseyburkhardt** [comment]
We encountered this today as well -- at HEAD

```
long millis = System.currentTimeMillis();
assertThat(DateUtils.isToday(millis)).isTrue();  // fails
```

**Comment by hoisie** [comment]
The problem is that the implementation of [DateUtils.isToday](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/text/format/DateUtils.java;l=522?q=DateUtils.isToday&ss=android) makes a call to `System.currentTimeMillis`. When `System.currentTimeMillis` is called by the framework, it it intercepted and the value returned is based on a virtual time that is determined by the scheduler. I think it has a bogus value such as 100. 

We could shadow DateUtils.isToday to use the JVM's implementation of System.currentTimeMillis, or perhaps adjust the starting value of the virtual time to be the current time. @brettchabot any thoughts?

**Comment by hoisie** [comment]
Another option is we could selectively intercept System.currentTimeMillis. For instance, we could probably disable it for Date/Time formatting functions, but enable it for scheduling-related functions. That is definitely doable.

**Comment by hoisie** [comment]
The current workaround is to do this:

https://github.com/robolectric/robolectric/issues/3912#issuecomment-459508750

**Comment by carstenhag** [comment]
As I have commented at #6658, with Robolectric 4.6.1 you have to use this line:
`ShadowSystemClock.advanceBy(Duration.ofSeconds(Instant.now().epochSecond))` and then the DateUtils methods work as expected.
